#!/usr/bin/env python
PACKAGE = "asl_turtlebot"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("k1",       double_t,    0, "Pose Controller k1", 0.8,  0., 2.0)
gen.add("k2",       double_t,    0, "Pose Controller k2", 0.4,  0., 2.0)
gen.add("k3",       double_t,    0, "Pose Controller k3", 0.4,  0., 2.0)
gen.add("spline_alpha",       double_t,    0, "Traj Smooth spline_alpha", 0.008, 0.0, 0.15)

gen.add("kpx",       double_t,    0, "Traj Controller kpx", 0.5,  0., 5.0)
gen.add("kpy",       double_t,    0, "Traj Controller kpy", 0.5,  0., 5.0)
gen.add("kdx",       double_t,    0, "Traj Controller kdx", 1.5,  0., 5.0)
gen.add("kdy",       double_t,    0, "Traj Controller kdy", 1.5,  0., 5.0)
gen.add("V_max",       double_t,    0, "Traj Controller V_max", 0.2,  0., 5.0)
gen.add("om_max",       double_t,    0, "Traj Controller om_max", 0.4,  0., 5.0)

gen.add("chunky_radius", double_t, 0, "Chunky Radius", 0.11, 0.10, 0.3)

exit(gen.generate(PACKAGE, "navigator", "Navigator"))
